libs/util/system.h

```diff
@@ -238,6 +239,20 @@ betoh (T const& x)
 #   error "Couldn't determine host endianess!"
 #endif
 
+template <typename T>
+T read_binary_little_endian(std::istream* stream) {
+  T data_little_endian;
+  stream->read(reinterpret_cast<char*>(&data_little_endian), sizeof(T));
+  return letoh(data_little_endian);
+}
+
+template <typename T>
```

Comment:
This is an odd function that modifies the input vector yet expects it to be initialized to a certain size. This also doesn't seem very versatile, and perhaps shouldn't be in system.h. Can you move that function to the place where you actually use it (to your other PR)?